packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -606,7 +606,10 @@ function walkExprTree(
         );
       }
 
-      walkExprTree(expr.__expr__, parentScope, ctx);
+      childExprs.push(
+        ...walkExprTree(expr.__expr__, parentScope, ctx),
+        ...insertChildExprs,
+      );
```

Comment:
I’m typing this out to ensure I understand the fix here:

In a previous with extraction fix, we extracted child expressions from the `unlessConflict` expression and stored them in the `insertChildExprs` array. However, we didn’t add these child expressions to the parent (`insert`) expression’s child expression list. Consequently, when the expression -> query machinery was validating the with access, it was not able to see that these expressions were allowed.